plugins/provisioners/salt/default_downloader.rb

```diff
@@ -0,0 +1,93 @@
+require 'open-uri'
+require 'digest'
+require_relative "./errors"
+
+module VagrantPlugins
+  module Salt
+    class DefaultDownloader
+      BASE_URL = "https://repo.saltproject.io/bootstrap/stable/"
+      PS1 = "bootstrap-salt.ps1"
+      SH  = "bootstrap-salt.sh"
+      SHA256_SUFFIX = "sha256"
+      SIG_SUFFIX = "asc"
+      KEY_URL = "#{BASE_URL}SALT-PROJECT-GPG-PUBKEY-2023.gpg"
+
+      def initialize(communicator)
```

Comment:
Should include passing in what the guest platform is for script determination (value in `config.vm.guest`)